src/ruby/pb/test/xds_client.rb

```diff
@@ -71,6 +88,29 @@ def create_stub(opts)
   )
 end
 
+class ConfigureTarget < Grpc::Testing::XdsUpdateClientConfigureService::Service
+  include Grpc::Testing
+
+  def configure(req, _call)
+    $rpcs_to_send = req['types'];
```

Comment:
We'd want updates for `$rpcs_to_send` and `$metadata_to_send` to be _atomic_. That is, the thread running `run_test_loop` should see them consistently, not interleave with this `configure` call (e.g., `$rpcs_to_send` is updated but `$metadata_to_send` is not).

In Java, we created a data structure ([`RpcConfig`](https://github.com/grpc/grpc-java/blob/499694e9da1982f5a23ae6ea8b1fa8b63f63a2c1/interop-testing/src/main/java/io/grpc/testing/integration/XdsTestClient.java#L508)) to hold `rpcs_to_send` and `metadata_to_send`. Each time this configure() RPC is invoked, it creates a new `RpcConfig` instance to replace the global reference.

In C-core, it is similar, but just slightly different representations (It uses [a queue of configs](https://github.com/grpc/grpc/blob/e33849682c410452596ebf008a2b984e388e4f99/test/cpp/interop/xds_interop_client.cc#L411-L419), each of which is the configuration for one RPC type).

Reply:
Fixed. Thanks for the suggestion. I have a `RpcConfig` object now.